### DISCUSSION TITLE
Virtual field hidden in create view when GraphQL set to omit on read

### DISCUSSION THREAD

**Discussion by linus-ha** [keystonejs/keystone#8501] [Questions]
### Steps to reproduce the problem
Add this field to a list:
```js
hint: virtual({
  field: graphql.field({
    type: graphql.Boolean,
    resolve(item) {return true}
  }),
  graphql: {omit: ["update", "read"]},
  ui: {
    views: './fields/helloWorld',
    //createView: { fieldMode: 'edit' },
    listView: { fieldMode: 'hidden' },
    itemView: { fieldMode: 'hidden' },
  },
})
```

using this custom view to be saved at `./fields/helloWorld.tsx`
```js
import { FieldProps } from '@keystone-6/core/types';
import { FieldContainer } from '@keystone-ui/fields';
import { controller } from '@keystone-6/core/fields/types/json/views';

export const Field = ({ value }: FieldProps<typeof controller>) => {
  return (
    <FieldContainer>
        <pre>
            Hello, World!
        </pre>
    </FieldContainer>
  );
};
```

When setting `omit` to `read` and/or `create`, the virtual field, better said "the corresponding view", is not shown. Omitting the field in GraphQL seems to result in not displaying the field in the overlay create view.

This occurs in the normal create view as well as in the overlay / inline create view.

### Please describe what you expected to happen
Show my custom view in the overlay ("inline") create view as configured via the ui parameters. At lest when setting `omit` to `["update", "read"]` the field (view) should be visible when **creating** an entry.

### Version / etc.
```
[linus@linus-pc cms]$ node -v
v18.12.1
```

```json
  "dependencies": {
    "@keystone-6/auth": "^5.0.1",
    "@keystone-6/core": "^3.1.3",
    "@keystone-6/fields-document": "^5.0.2",
    "typescript": "^4.9.3"
  },
  "engines": {
    "node": "^18.12"
  }
```

**Comment by borisno2** [comment]
Hi @linus-ha 
Thanks for posting, I am interested in what you ware wanting to achieve here if you are happy to expand. Are you only wanting to display this field in the AdminUI on Create and no data is associated with it - ie it is a static componenet? 

The purpose of the Virtual Field is to be a GraphQL read-only field, ie to get/transform data and have it accessed through GraphQL. It only has a GraphQL `read`, so `omit: ['read']` is removing it from GraphQL entirely, including from our internal admin meta - ie if keystone can't read it, it can't display it, even on create.

**Comment by linus-ha** [comment]
Hi, exactly. As the field is called, this should be a hint of unstructured data of an external data source from which we need to create internal Items. This "hint" component will gather some information from the parent entry visible in the background and therefore present in the DOM, as this will only be displayed when clicking on the inline creation field, which results in a dialog being shown that is displayed above my parent entry. I am doing that, so that the user can access that information from within the creation view (e.g. copy text, do some research and enter correct Information instead).

![image](https://user-images.githubusercontent.com/30652341/208790541-8b9de36c-d0b6-4fb9-a851-c6e4e7d42ea8.png)


I thought so, that the UI probably represents its GraphQL equivalent. That's why I thought not omitting `create` would not affect the availability of that field in the create view, but as you've said, it does.

If you have any better idea of how to render a component in the create view only while not being exposed via GraphQL as a field, I would of course implement that instead. As it is currently, it integrates well with in the schema configuration.
In addition: Is there any possibility in passing data from the parent entry to the inline crate view that I do not know of yet? Reading data from the DOM is rather ugly.

**Comment by pc-erin** [comment]
@linus-ha This might have to wait until the "Next-gen Admin UI" section on the roadmap starts. That'll probably allow the customization you want.

There's [custom field views](https://github.com/keystonejs/keystone/tree/main/examples/custom-field-view) that let you make  your own interactive component on the admin panel, but I don't think they have access to the values of other fields.
